plugins/joblib_launcher/tests/test_joblib_launcher.py

```diff
@@ -0,0 +1,54 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+import pytest
+
+from hydra.core.plugins import Plugins
+from hydra.plugins.launcher import Launcher
+from hydra.test_utils.launcher_common_tests import (
+    IntegrationTestSuite,
+    LauncherTestSuite,
+)
+
+# This has to be included here for the LauncherTestSuite to work.
+# noinspection PyUnresolvedReferences
+from hydra.test_utils.test_utils import sweep_runner  # noqa: F401
+from hydra_plugins.joblib_launcher import JoblibLauncher
+
+
+def test_discovery() -> None:
+    # Tests that this plugin can be discovered via the plugins subsystem when looking for Launchers
+    assert JoblibLauncher.__name__ in [x.__name__ for x in Plugins.discover(Launcher)]
+
+
+@pytest.mark.parametrize("launcher_name, overrides", [("joblib", [])])
+class TestJoblibLauncher(LauncherTestSuite):
+    """
+    Run the Launcher test suite on this launcher.
+    Note that hydra/launcher/joblib.yaml should be provided by this launcher.
+    """
+
+    pass
+
+
+@pytest.mark.parametrize(
+    "task_launcher_cfg, extra_flags, plugin_module",
+    [
+        (
+            {
+                "defaults": [
+                    {"hydra/launcher": "joblib"},
+                    {"hydra/hydra_logging": "hydra_debug"},
```

Comment:
I see this is was inherited from the example launcher. It's probably not required.
did you find it helpful?

Reply:
Changed this to test with process-based (default config) and thread-based parallelism in the latest commit

Reply:
This is good, but could actually be related to your crash.
Maybe there is some cleanup to do after using the thread based method?